fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java

```diff
@@ -121,89 +133,77 @@ public String getLatestVersion(String url) {
      */
-    public boolean hasResource(String url) {
-        if (url == null) {
-            return false;
-        }
-
-        String id = null;
-        int index = url.indexOf("#");
-        if (index != -1) {
-            id = url.substring(index + 1);
-            url = url.substring(0, index);
-        }
-
-        String version = null;
-        index = url.indexOf("|");
-        if (index != -1) {
-            version = url.substring(index + 1);
-            url = url.substring(0, index);
-        }
-
-        FHIRRegistryResource resource = findResource(url, version);
-        return (id != null) ? (getResource(resource, url, id) != null) : (resource != null);
+    public <T extends Resource> Collection<T> getResources(Class<T> resourceType) {
+        Objects.requireNonNull(resourceType);
+        return providers.stream()
+                .map(provider -> provider.getRegistryResources(resourceType))
+                .flatMap(Collection::stream)
+                .map(registryResource -> resourceType.cast(registryResource.getResource()))
+                .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
     }
 
-
     /**
-     * Unload the resource associated with the given url.
+     * Get the profiles that constrain the given type as a collection of {@link Canonical} URLs
      *
-     * @param url
-     *     the url of the resource
+     * @param type
+     *     the constrained type
+     * @return
+     *     the profiles that constrain the given type as a collection of {@link Canonical} URLs
      */
-    public void unloadResource(String url) {
-        if (url == null) {
-            return;
-        }
-
-        int index = url.indexOf("#");
-        if (index != -1) {
-            url = url.substring(0, index);
-        }
-
-        String version = null;
-        index = url.indexOf("|");
-        if (index != -1) {
-            version = url.substring(index + 1);
-            url = url.substring(0, index);
-        }
+    public Collection<Canonical> getProfiles(String type) {
+        Objects.requireNonNull(type);
+        return providers.stream().map(provider -> provider.getProfileResources(type))
+                .flatMap(Collection::stream)
+                .map(registryResource -> Canonical.of(registryResource.getUrl(), registryResource.getVersion().toString()))
+                .sorted(CANONICAL_COMPARATOR)
+                .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
+    }
 
-        FHIRRegistryResource resource = findResource(url, version);
-        if (resource != null) {
-            resource.unload();
-        }
+    /**
+     * Get the search parameters with the given search parameter type (e.g. string, token, etc.)
+     *
+     * @param type
+     *     the search parameter type
+     * @return
+     *     the search parameters with the given search parameter type
+     */
```

Comment:
maybe add a reference to `getResources(Class<T> resourceType)` to make it clear how one would get all search parameters (irregardless of type)

Reply:
added